src/packet/signature.js

```diff
@@ -407,23 +406,27 @@ class SignaturePacket {
     return util.concat([length, result]);
   }
 
-  // V4 signature sub packets
+  // Signature subpackets
   readSubPacket(bytes, hashed = true) {
     let mypos = 0;
 
     // The leftmost bit denotes a "critical" packet
     const critical = !!(bytes[mypos] & 0x80);
     const type = bytes[mypos] & 0x7F;
 
+    mypos++;
+
     if (!hashed) {
-      this.unhashedSubpackets.push(bytes.subarray(mypos, bytes.length));
+      this.unhashedSubpackets.push({
+        type,
+        critical,
+        body: bytes.subarray(mypos, bytes.length)
+      });
       if (!allowedUnhashedSubpackets.has(type)) {
         return;
```

Comment:
Not relevant for the PR, but we might want to log this as a debug error...?

Reply:
Yeah, we could, though arguably it's not really an error, as storing stuff in the unhashed subpackets is allowed; it just shouldn't be trusted. But there could be semi-legitimate cases for e.g. a keyserver to store some information there if it trusts its own storage, for example (though it could also store it externally to the key, of course).

Very pedantically, it could also be argued that this variable should be called `trustableUnhashedSubpackets` instead of `allowedUnhashedSubpackets` ^.^